EIPS/eip-draft_request_id.md

```diff
@@ -0,0 +1,155 @@
+
+## Motivation
+<!--The motivation is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. EIP submissions without sufficient motivation may be rejected outright.-->
+The lack of request identifiers in the request / response paris of the `eth` protocol puts unnecessary burden of code complexity into every Ethereum client. It also makes the communication slightly less efficient. Another argument can be made that the addition of request identifiers makes the protocol more aligned with the `les` protocol which **does** already defines request identifiers for each request / response pair.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+
+Change the following message types in the `eth` protocol:
+
+* `GetBlockHeaders (0x03)`
+   * **Current:** `[block: {P, B_32}, maxHeaders: P, skip: P, reverse: P in {0, 1}]`
+   * **Then**: `[request_id: P, [block: {P, B_32}, maxHeaders: P, skip: P, reverse: P in {0, 1}]]`
+* `BlockHeaders (0x04)`
+   * **Current:** `[blockHeader_0, blockHeader_1, ...]`
+   * **Then**: `[request_id: P, [blockHeader_0, blockHeader_1, ...]]`
+* `GetBlockBodies (0x05)`
+   * **Current:** `[hash_0: B_32, hash_1: B_32, ...]`
+   * **Then**: `[request_id: P, [hash_0: B_32, hash_1: B_32, ...]]`
+* `BlockBodies (0x06)`
+   * **Current:** `[hash_0: B_32, hash_1: B_32, ...]`
+   * **Then**: `[request_id: P, [hash_0: B_32, hash_1: B_32, ...]]`
+* `GetNodeData (0x0d)`
+   * **Current:** `[hash_0: B_32, hash_1: B_32, ...]`
+   * **Then**: `[request_id: P, [hash_0: B_32, hash_1: B_32, ...]]`
+* `NodeData (0x0e)`
+   * **Current:** `[value_0: B, value_1: B, ...]`
+   * **Then**: `[request_id: P, [value_0: B, value_1: B, ...]]`
+* `GetReceipts (0x0f)`
+   * **Current:** `[blockHash_0: B_32, blockHash_1: B_32, ...]`
+   * **Then**: `[request_id: P, [blockHash_0: B_32, blockHash_1: B_32, ...]]`
+* `Receipts (0x10)`
+   * **Current:** `[[receipt_0, receipt_1], ...]`
+   * **Then**: `[request_id: P, [[receipt_0, receipt_1], ...]]`
+
+To elaborate, each command is altered in the following way:
+
+1. Create a list with the `request_id` being the first element.
+2. Make the second element the list that defines the whole command in the current scheme.
+
+This is consistent with the request / response pairs in the `les` protocol.
+
+The ``request_id`` has the following characteristics:
+
+* 64 bit integer
+* Doesn't need to be sequential (can be random)
+* Does allow duplicates
+
+## Rationale
```

Comment:
I love this Q&A section! Are these questions you've already gotten, or are you predicting that these are the kinds of questions people will ask?

Reply:
A bit of both. Notice that I have posted this EIP now in the real repo: https://github.com/ethereum/EIPs/pull/2481

But the QA section is inspired by other EIPs which do that in the *rational* section, I thought it's a nice fit (overall, I struggle with the pre-defined sections and their meaning. I feel like repeating myself but this QA style helped me at least for this section)